### ISSUE TITLE
Can't build 0.2.0

### ISSUE THREAD

**Issue by aladairmaxwell** [MasterEric/Kade-Engine#25]
So I downloaded this 0.2.0 from `develop` branch and I can't build the game. First it required me to install 3 new libraries, and even after I did it, the console still giving me lots of errors. How do I compile newest version of Enigma Engine?

![image](https://user-images.githubusercontent.com/69295711/135397296-0095f2c5-a66e-4b47-b66b-2d838281116a.png)

**Comment by EliteMasterEric** [comment]
Hey,

If you just want to work with a built copy of the Enigma Engine (which has full support for the mods folder functionality documented on the wiki), use the pre-built binaries provided on the v0.2.0 pre-release: https://github.com/EnigmaEngine/EnigmaEngine/releases/tag/v0.2.0-prerelease

If you want to build from scratch, I recommend you perform the steps on this wiki page:

**[Building Enigma Engine](https://github.com/EnigmaEngine/EnigmaEngine/wiki/building-enigma-engine)**

I try to keep that page up-to-date with each dependency I add to Enigma. Make sure to follow it thoroughly, even if you've already built Kade Engine before; in particular, note that:

* The Visual Studio dependency is the same but you'll likely need to re-install the Haxe dependencies.
* There's a bug with one of the debug tools, as you can see at `Enigma.hx:35`. I'll look into fixing it, but for now you NEED to use `git clone` in the command line to initialize the repository (as specified in the guide), and you can't just download the source code as a ZIP.
  * By the way, that's the only error in your log, the others are warnings or notices.
* The state of the `develop` branch is constantly changing as I work on stuff, and isn't guaranteed to be functional, but specific tags where I create releases are. To check out a specific tag, run `git checkout tag/v0.2.0-prerelease` before building.

**Comment by aladairmaxwell** [comment]
I did a copy using git but I don't understand how to get prerelease version, it says something like this. I don't actually get how git works, so I need some help >_<

![image](https://user-images.githubusercontent.com/69295711/135438321-6a859f2e-44c5-4ca2-9414-f62c400c90be.png)

**Comment by EliteMasterEric** [comment]
It looks like I made a typo in the command I gave you, run this instead:

```git commit -c tags/v0.2.0-prerelease```

**Comment by aladairmaxwell** [comment]
I made it and it works, thanks. But compiling error still exist.

![image](https://user-images.githubusercontent.com/69295711/135531581-bb1fb94b-7f82-4062-b0c9-0bc45d5b11b7.png)

**Comment by EliteMasterEric** [comment]
ugh hmm let me try again

Okay I figured it out:

```
git switch -c tags/v0.2.0-prerelease
```

Normally I use a graphical program like GitKraken for this stuff...

**Comment by aladairmaxwell** [comment]
Same error. I want to compile debug version, but I can't even create regular one. 

![image](https://user-images.githubusercontent.com/69295711/135579196-07e49916-13cd-4e83-8024-57c7bbd37fdb.png)

**Comment by EliteMasterEric** [comment]
Looks like you're using a wrong version of the Polymod dependency.

> 
> If you want to build from scratch, I recommend you perform the steps on this wiki page:
> 
> **[Building Enigma Engine](https://github.com/EnigmaEngine/EnigmaEngine/wiki/building-enigma-engine)**
> 
> I try to keep that page up-to-date with each dependency I add to Enigma. Make sure to follow it thoroughly, even if you've already built Kade Engine before; in particular, note that:
> 
>     * The Visual Studio dependency is the same but **you'll likely need to re-install the Haxe dependencies.**

**Comment by aladairmaxwell** [comment]
I decided to delete everything and clone git from clean start. I used these commands, but the code still does not want to switch to prerelease, it uses stable code. There are no transformations about this. I can switch to develop like this as well and it works. How about to create a new branch with prerelease code then?

![image](https://user-images.githubusercontent.com/69295711/135685883-fc611e67-778b-4509-a98e-38e1dee966d6.png)

**Comment by EliteMasterEric** [comment]
If you think working with the command line isn't worth your time, I suggest checking out the free program [GitKraken](https://www.gitkraken.com/). It is a graphical program used to manage Git repositories. I have been using it myself for years to great effect, and it should easily be able to figure out what your intention is if you click a specific tag and try to checkout.

If you want to persist with the command line, read on.

The `git commit` command is not what you want at all, that's related to writing code to the repo. Part of your problem appears that you are likely confusing specific commands and mixing and matching command line options. I did some research and the `git switch` command has some useful syntax.

`git switch -c tags/v0.2.0-release` is saying to create a new branch unconditionally, which will fail if you've made a branch with that name.

`git switch -guess tags/v0.2.0-release` should first check if there is a local branch and switch to that if there is, or use the remote if there isn't. This one should work.

Or you can just use GitKraken (or do your research and choose another GUI if you like).

**Comment by aladairmaxwell** [comment]
Alright, thanks, I used a GitKraken and it seems fine. It's a shame that the command lines weren't working as I expected. But now it's ok. I can build now. Thanks once again!
